airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/AirbyteType.java

```diff
@@ -0,0 +1,76 @@
+package io.airbyte.integrations.destination.bigquery.typing_deduping;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public sealed interface AirbyteType permits AirbyteType.Primitive, AirbyteType.Object, AirbyteType.Array, AirbyteType.OneOf {
```

Comment:
I think this file is secretly being opinionated about the protocol itself... possibly this is supposed to live in airbyte-protocol